### PR TITLE
add edit update stats

### PR DESCRIPTION
**Pull request by ZappoMan**
This adds some debugging stats to the Entity Statistics window in the client. 

New stats include:
* Average "transit time" between when an entity is edited on some client until the viewing client sees the change in its local environment. 
* Max transit time. 
* Rolling average of updates per second, total edits, and average size in bytes of edits.
* Rolling average of number of tree elements per packet and per second.
* Rolling average of number of entities updated per packet and per second.
* Rolling average for processing time, uncompress time, and lock wait time per packet.

### CODE REVIEW

File: interface/src/devices/TV3DManager.cpp [hunk 1]
```diff
@@ -109,7 +109,6 @@ void TV3DManager::display(RenderArgs* renderArgs, Camera& whichCamera) {
         glScissor(portalX, portalY, portalW, portalH);
 
         glm::mat4 projection = glm::frustum<float>(eye.left, eye.right, eye.bottom, eye.top, nearZ, farZ);
-        float fov = atan(1.0f / projection[1][1]);
```
Comment (ZappoMan) [inline interface/src/devices/TV3DManager.cpp]: warning cleanup

File: interface/src/ui/ApplicationCompositor.cpp [hunk 2]
```diff
@@ -25,18 +25,13 @@
 
 
 // Used to animate the magnification windows
-static const float MAG_SPEED = 0.08f;
```
Comment (ZappoMan) [inline interface/src/ui/ApplicationCompositor.cpp]: warning cleanup

File: interface/src/ui/AvatarInputs.cpp [hunk 3]
```diff
@@ -67,7 +67,6 @@ void AvatarInputs::update() {
     AI_UPDATE(cameraMuted, Menu::getInstance()->isOptionChecked(MenuOption::MuteFaceTracking));
 
     auto audioIO = DependencyManager::get<AudioClient>();
-    const float CLIPPING_INDICATOR_TIME = 1.0f;
```
Comment (ZappoMan) [inline interface/src/ui/AvatarInputs.cpp]: warning cleanup

File: interface/src/ui/OctreeStatsDialog.cpp [hunk 4]
```diff
@@ -203,6 +204,20 @@ void OctreeStatsDialog::paintEvent(QPaintEvent* event) {
         "Leaves: " << qPrintable(serversLeavesString) << "";
     label->setText(statsValue.str().c_str());
 
+    label = _labels[_entityUpdateTime];
```
Comment (ZappoMan) [inline interface/src/ui/OctreeStatsDialog.cpp]: this is the UI in the stats dialog box

File: libraries/entities/src/EntityItem.cpp [hunk 5]
```diff
@@ -414,6 +414,14 @@ int EntityItem::readEntityDataFromBuffer(const unsigned char* data, int bytesLef
     dataAt += sizeof(lastEditedFromBuffer);
     bytesRead += sizeof(lastEditedFromBuffer);
     lastEditedFromBufferAdjusted = lastEditedFromBuffer - clockSkew;
+
+    // Tracking for editing roundtrips here. We will tell our EntityTree that we just got incoming data about
+    // and entity that was edited at some time in the past. The tree will determine how it wants to track this
+    // information.
+    if (_element && _element->getTree()) {
```
Comment (ZappoMan) [inline libraries/entities/src/EntityItem.cpp]: here's where we get incoming changes, and we send it off to the EntityTree to track it or not.

File: libraries/entities/src/EntityTree.cpp [hunk 6]
```diff
@@ -1104,3 +1107,28 @@ bool EntityTree::readFromMap(QVariantMap& map) {
...
+    _totalTrackedEdits = 0;
+}
+
+
+
+void EntityTree::trackIncomingEntityLastEdited(quint64 lastEditedTime) {
+    // we don't want to track all edit deltas, just those edits that have happend
+    // since we connected to this domain. This will filter out all previously created
+    // content and only track new edits
+    if (lastEditedTime > _treeResetTime) {
```
Comment (ZappoMan) [inline libraries/entities/src/EntityTree.cpp]: we don't track edits that are older than our reset time. As the comment explains this prevents including non-realtime edits

**Comment by hifi-gustavo** [comment]
Build console output for commit: b13a4b2a71d6b027ae109c06ba2aeae85b5e7596 available at: https://gist.github.com/89dd1ce1b766687dfd6b<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5253.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5253.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 63b585da0fda9cccd69453513c5ba545c39e44e3 available at: https://gist.github.com/89dd1ce1b766687dfd6b<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5253.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5253.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5c4d4893def43ca46c10f39eef2a0ce38c4662d5 available at: https://gist.github.com/89dd1ce1b766687dfd6b<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5253.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5253.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: b568e895ec6e8852692005046645b033ec1d94fb available at: https://gist.github.com/89dd1ce1b766687dfd6b<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5253.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5253.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0c0b8dbbc15c0255abc06453a596ca1b9e17b8f1 available at: https://gist.github.com/89dd1ce1b766687dfd6b<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5253.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5253.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8916dd205664761e2052a09562eade21c2e0f889 available at: https://gist.github.com/89dd1ce1b766687dfd6b<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5253.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5253.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: a54cf6706470b28b10e7feb539cc5a254bd055d4 available at: https://gist.github.com/89dd1ce1b766687dfd6b<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5253.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5253.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: e1530b3ce942fa2f19b730089d81fe5de02309ac available at: https://gist.github.com/89dd1ce1b766687dfd6b<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5253.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5253.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2aa345dfc6a9f061f2126ad35be5a734c8bce265 available at: https://gist.github.com/89dd1ce1b766687dfd6b<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5253.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5253.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 40cdd8646e3c727fdbb558fd618ed231a39e274b available at: https://gist.github.com/89dd1ce1b766687dfd6b<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5253.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5253.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 74733452da2cdda7ead883bb680983f0ac809df2 available at: https://gist.github.com/89dd1ce1b766687dfd6b<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5253.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5253.dmg

File: libraries/octree/src/OctreeRenderer.cpp [hunk 7]
```diff
@@ -134,17 +148,57 @@ void OctreeRenderer::processDatagram(const QByteArray& dataByteArray, const Shar
...
+        _uncompressPerPacket.updateAverage(totalUncompress);
+        _readBitstreamPerPacket.updateAverage(totalReadBitsteam);
+        
+        quint64 now = usecTimestampNow();
+        if (_lastWindowAt == 0) {
+            _lastWindowAt = now;
+        }
+        quint64 sinceLastWindow = now - _lastWindowAt;
+        
+        const quint64 USECS_PER_SECOND = 1000 * 1000;
```
Comment (Atlante45) [inline libraries/octree/src/OctreeRenderer.cpp]: We already have that constant in NumericalConstants.h

**Comment by hifi-gustavo** [comment]
Build console output for commit: c4a5160b890260f75f21e99e88e6f9e8267fc43e available at: https://gist.github.com/89dd1ce1b766687dfd6b<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5253.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5253.dmg

**Pull request merged**